assets/js/data/collections/resolvers.js

```diff
@@ -44,6 +60,14 @@ export function* getCollection( namespace, resourceName, query, ids ) {
 			headers,
 		} = yield apiFetchWithHeaders( route + queryString );
 
+		if ( headers && headers.get && headers.has( 'last-modified' ) ) {
+			// Do any invalidation before the collection is received to prevent
+			// this query running again.
+			yield invalidateModifiedCollection(
+				parseInt( headers.get( 'last-modified' ), 10 )
+			);
+		}
+
```

Comment:
Will test to verify, but one thing I like about this, is the state should get cleared _before_ the new collection is saved to the state. So what's really cool about this is we don't end up doing another network request for the collection just retrieved. Great thinking!

Reply:
So after testing, it appears that it's not behaving like I expected where the _current_ query only resolves once. I think we can deal with this in a performance/optimization follow-up (along with the initial products request happening twice on page load, which happens in master too).

Reply:
@nerrad I noticed this and couldn't see an obvious way of invalidating everything but the current request. Any pointers before I merge this, or I can just create followup.

Reply:
Let's just do it as a follow-up. The important pieces are in this pull and we can iterate on performance outside of it.

Reply:
Although one immediate idea comes to mind, and that is right after invalidating, you dispatch a `finishResolution` action for the current resolver (with it's args) and that _should_ in theory prevent that resolver from being called again. Still, it can be tried in a follow up.

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1765